parlai/crowdsourcing/tasks/acute_eval/acute_eval_blueprint.py

```diff
@@ -35,123 +89,31 @@ class AcuteEvalBlueprint(Blueprint):
     AgentStateClass: ClassVar[Type["AgentState"]] = AcuteEvalAgentState
     TaskBuilderClass: ClassVar[Type["TaskBuilder"]] = AcuteEvalBuilder
     TaskRunnerClass: ClassVar[Type["TaskRunner"]] = AcuteEvalRunner
+    ArgsClass = AcuteEvalBlueprintArgs
     supported_architects: ClassVar[List[str]] = ["mock"]  # TODO update
     BLUEPRINT_TYPE = BLUEPRINT_TYPE
 
     @classmethod
-    def assert_task_args(cls, opts: Any) -> None:
+    def assert_task_args(cls, args: DictConfig, shared_state: SharedTaskState) -> None:
         """
         Ensure that the data can be properly loaded.
         """
-        if opts.get("pairings_filepath") is not None:
-            pairings_filepath = os.path.expanduser(opts["pairings_filepath"])
+        if args.blueprint.get("pairings_filepath") is not None:
+            pairings_filepath = os.path.expanduser(args.blueprint.pairings_filepath)
             assert os.path.exists(
                 pairings_filepath
             ), f"Provided file {pairings_filepath} doesn't exist"
-        elif opts.get("pairings_task_data") is not None:
-            assert (
-                len(opts.get("pairings_task_data")) > 0
-            ), "Length of data dict provided was 0"
         else:
             raise AssertionError(
                 "Must provide one of a data csv, json, or a list of tasks"
             )
 
-        if opts.get("block_on_onboarding_fail") is True:
-            if opts.get("block_qualification") is None:
+        if args.blueprint.get("block_on_onboarding_fail") is True:
```

Comment:
If you do provide a default value however, you can just access directly without running the risk of crashing due to the variable not being set. `args.blueprint.block_on_onboarding_fail` works here.

You already do this elsewhere, so this one is just a nit really.

Reply:
Hmm, but we call `block_on_onboarding_fail` on the first line here and `block_qualification` on the second line, so we still need to check if `block_on_onboarding_fail` is set

Reply:
But in the dataclass you provide a default value:
```
block_on_onboarding_fail: bool = field(
    default=True, metadata={"help": "whether to block on onboarding failure"}
)
```
Due to this, unless the user is doing something that they really shouldn't be (like using a configuration for this that doesn't derive from your `AcuteEvalBlueprintArgs`, this should always be set.

Realistically though this doesn't need to change, I was more using it as a discussion point to convey some of how hydra defaults work.

Reply:
Ah, yes, I see what you mean that a default value has been set here - yes, it's good in that case to assume that the attribute does exist. Changed